explorer/apps.py

```diff
@@ -51,12 +51,14 @@ def track_summary_stats():
     from explorer.tracker import gather_summary_stats
     from explorer.models import Query
 
-    # Test to see if migrations have been applied.
-    # If not, then we won't gather start-up stats (no need -- there's nothing interesting to report on)
-    # We can't use a post_migrate hook because in many cases the app will be starting
-    # and there won't be any migrations running! So the signal would never get called.
-    # Django doesn't actually have a way of running code on application initialization,
-    # so we are kinda faking it by doing this.
+    # Django doesn't actually have a way of running code on application initialization, so we have come up with this.
+    # The app.ready() method (the call site for this function) is invoked *before* any migrations are run. So if were
```

Comment:
It sounds like maybe you want to have a signal reciever that runs on the `post_migrate` signal.

https://docs.djangoproject.com/en/5.0/ref/signals/#post-migrate

I did something similar for django-cms where it uses `pre_migrate` to prevent people from accidentally migrating their database;
https://github.com/django-cms/django-cms/blob/develop-4/cms/signals/__init__.py#L35

Reply:
No -- keep reading the comment. I tried that, but if there are no actual migrations to run, then post_migrate never gets called. The intent of this is to grab summary stats on every application start-up -- many of which will not have net-new migrations and therefore never fire the signal.